pointer_test.go

```diff
@@ -94,6 +97,22 @@ func TestPointer(t *testing.T) {
 				atom := tt.newAtomic()
 				require.Equal(t, fmt.Sprint(tt.initial), atom.String(), "String did not return the correct value.")
 			})
+
+			t.Run("MarshalJSON", func(t *testing.T) {
+				atom := tt.newAtomic()
+				marshaledPointer, err := json.Marshal(atom)
+				require.NoError(t, err)
+				marshaledRaw, err := json.Marshal(tt.initial)
+				require.NoError(t, err)
+				require.Equal(t, marshaledRaw, marshaledPointer, "MarshalJSON did not return the correct value.")
+			})
 		})
 	}
+
```

Comment:
Could a test case be added for the Unmarshal case?

Reply:
not sure what are you asking, do you mean line 112?

Reply:
Ok, I get it. you mean Unmarshal error case